parquet-hadoop/src/main/java/org/apache/parquet/hadoop/codec/CleanUtil.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.hadoop.codec;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.ByteBuffer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CleanUtil {
+  private static final Logger logger = LoggerFactory.getLogger(CleanUtil.class);
+  private static final Field CLEANER_FIELD;
+  private static final Method CLEAN_METHOD;
+
+  static {
+    ByteBuffer buf = null;
+    try {
+      buf = ByteBuffer.allocateDirect(1);
+      CLEANER_FIELD = buf.getClass().getDeclaredField("cleaner");
+      CLEANER_FIELD.setAccessible(true);
+      Object cleaner = CLEANER_FIELD.get(buf);
+      CLEAN_METHOD = cleaner.getClass().getDeclaredMethod("clean");
+    } catch (NoSuchFieldException | NoSuchMethodException | IllegalAccessException e) {
+      clean(buf);
```

Comment:
Cannot see the purpose of invoking clean in case of the field or the method was not properly initialized.

Reply:
Yes, you are right, here the `clean` may not do real work for `CLEAN_METHOD` and `CLEAN_FIELD` is not available, but I think keep the behavior consistent is necessary.

> buf = ByteBuffer.allocateDirect(1);  // allocated memory, then we should release it when exception happens.

And also it can print some message about the memory leak, though, it's non big deal.

Reply:
If you only want to be consistent then you should clean the buffer in finally instead in a catch block. Currently, it only cleans the buffer if an error occurs but doesn't if everything goes normally.

Reply:
@gerashegalov fixed! Thanks!